pkg/operator/ceph/controller/spec.go

```diff
@@ -653,8 +655,27 @@ func PrivilegedContext(runAsRoot bool) *v1.SecurityContext {
 	return sec
 }
 
-// LogCollectorContainer runs a cron job to rotate logs
+// LogCollectorContainer rotate logs
 func LogCollectorContainer(daemonID, ns string, c cephv1.ClusterSpec) *v1.Container {
+
+	maxLogSize := "500M"
+	if c.LogCollector.MaxLogSize != nil {
+		maxLogSize = fmt.Sprintf("%dM", c.LogCollector.MaxLogSize.Value()/1000/1000)
+		if maxLogSize == "0M" {
+			logger.Info("maxLogSize is 0M setting to minimum of 1M")
+			maxLogSize = "1M"
+		}
+	}
+
+	periodicity := "daily"
+	if c.LogCollector.Periodicity == "1h" || strings.Contains(c.LogCollector.Periodicity, "h") {
```

Comment:
added `|| strings.Contains(c.LogCollector.Periodicity, "h")` this will help during upgrade. Also, test
before my changes, I had `2h` and with my changes, it was changed to `hourly` which accepted the format for logrotate.

@travisn @BlaineEXE

Reply:
Ok thanks for confirming that during upgrade the previous values are still allowed. I would actually suggest that the check for containing "h" should be moved under daily. In other words, any hourly value besides "1h" will default to daily in the new approach.

Reply:
```suggestion
	if c.LogCollector.Periodicity == "1h" {
```